src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -1737,60 +1737,59 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
-            e.GetTargetType() == SMART_TARGET_CLOSEST_ENEMY || e.GetTargetType() == SMART_TARGET_CLOSEST_FRIENDLY ||
-            e.GetTargetType() == SMART_TARGET_SELF || e.GetTargetType() == SMART_TARGET_STORED)) */
+            switch (e.GetTargetType())
             {
-                // we want to move to random element
-                if (!targets.empty())
-                    target = Acore::Containers::SelectRandomContainerElement(targets);
-            }
-
-            if (!target)
+            case SMART_TARGET_POSITION:
             {
                 G3D::Vector3 dest(e.target.x, e.target.y, e.target.z);
                 if (e.action.moveToPos.transport)
                     if (TransportBase* trans = me->GetDirectTransport())
                         trans->CalculatePassengerPosition(dest.x, dest.y, dest.z);
 
                 me->GetMotionMaster()->MovePoint(e.action.moveToPos.pointId, dest.x, dest.y, dest.z, true, true,
-                                                 isControlled ? MOTION_SLOT_CONTROLLED : MOTION_SLOT_ACTIVE, e.target.o);
+                    isControlled ? MOTION_SLOT_CONTROLLED : MOTION_SLOT_ACTIVE, e.target.o);
+
+                break;
             }
-            else // Xinef: we can use dest.x, dest.y, dest.z to make offset
+            case SMART_TARGET_RANDOM_POINT:
+            if (me)
             {
+                float range = (float)e.target.randomPoint.range;
+                Position srcPos = { e.target.x, e.target.y, e.target.z, e.target.o };
+                Position randomPoint = me->GetRandomPoint(srcPos, range);
+                me->GetMotionMaster()->MovePoint(
+                    e.action.moveToPos.pointId,
+                    randomPoint.m_positionX,
+                    randomPoint.m_positionY,
+                    randomPoint.m_positionZ,
+                    true,
+                    true,
+                    isControlled ? MOTION_SLOT_CONTROLLED : MOTION_SLOT_ACTIVE
+                );
+
+                break;
+            }
+            // Can use target floats as offset
+            default:
+            {
+                // we want to move to random element
+                if (targets.empty())
+                    return;
+                else
+                    target = Acore::Containers::SelectRandomContainerElement(targets);
+
                 float x, y, z;
                 target->GetPosition(x, y, z);
                 if (e.action.moveToPos.ContactDistance > 0)
                 {
                     target->GetNearPoint(me, x, y, z, e.action.moveToPos.ContactDistance, 0, target->GetAngle(me));
                 }
                 me->GetMotionMaster()->MovePoint(e.action.moveToPos.pointId, x + e.target.x, y + e.target.y, z + e.target.z, true, true, isControlled ? MOTION_SLOT_CONTROLLED : MOTION_SLOT_ACTIVE);
+
+                break;
             }
+            }
```

Comment:
double curly

Reply:
first one is for the `default:`, second one is for the switch, I don't know what to do here in this case :s
![image](https://github.com/azerothcore/azerothcore-wotlk/assets/100873791/2fbab5ea-eee2-41f6-a81b-4421d6104da3)

Reply:
we could add 4 spaces for all lines inside the switch brackets I think

Reply:
agree

Reply:
Its just the ident that is broken (as per the review before)